### DISCUSSION TITLE
Electronic Sthetoscope using esp32 a1s audio development kit

### DISCUSSION THREAD

**Discussion by paralogicdr** [pschatzmann/arduino-audiokit#30] [Q&A] [answered]
Hi everyone,
I'm a cardiologist and new to esp32 programming. 
 I want to make a digital sthetoscope. I plan to record not only heart sounds but also simultaneously accuired ECG and PPG signals.
The accuired audio signals and two channel sensor signals will be streamed to a mobilephone and it will view and record signals
 I ordered this product as ECG and PPG sensor Protocentral's max86150 based sensor
https://protocentral.com/product/protocentral-max86150-ppg-and-ecg-breakout-with-qwiic-v2/
The sensor uses i2c protocol, which exists in 
Esp32 a1s audio development kit as gpio 21 and 22.  
The kit also has microphones, which I need.

I need your help about my questions if you'd like to help me:

Can I stream the sensor data and audio with acceptable quality simultaneously with esp32 a1s? If it is possible, how can I do this?
Heart sound frequencies are generally between 10- 1000 Hz. Should I use a hardware filter before codec chip?

**Comment by pschatzmann** [comment]
If you are only interrested in 10-1000 Hz you can use the smallest sampling rate that these boards support which is 8000 Hz. 
So I would expect that you can do both audio and signal processing together on the same board.

If you use the sound only for making it audible, and you don’t use it for further signal processing, I would expect that no filter is needed and you can plan to add a software low pass filter as a future improvement.

I never did any I2C on an AudioKit, so I am not sure about any obstacles there. 
Any further information on this would be welcome!

**Reply by paralogicdr** [reply]
Thank you Phil, it is important for me to record data from i2c and audio simultaneously. For further Signal processing I plan to use pc.

**Comment by pschatzmann** [comment, marked as answer]
No worries. Just start simple and if you run into limitations you still have plenty of options to improve:
- start a separate task for I2C and for audio
- use separate cores for each task

My major worry is I2C since I heard that some people were struggling...

**Reply by paralogicdr** [reply]
Thank you Phil, İ'll try
